src/GRANULAR/granular_model.h

```diff
@@ -0,0 +1,114 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   LAMMPS development team: developers@lammps.org
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#ifndef LMP_GRANULAR_MODEL_H
+#define LMP_GRANULAR_MODEL_H
+
+#include "pointers.h"    // IWYU pragma: export
+
+namespace LAMMPS_NS {
+namespace Granular_NS {
+
+#define EPSILON 1e-10
```

Comment:
This either should not be a define but a `static constexpr double` type or not in the header, but only in the .cpp files.

Reply:
Looking at this again, I don't think EPSILON should even be defined at this level but rather in the specific submodel files its used. Presumably one could define a separate force model that uses an epsilon-like value other than 1e-10. I'll move it to gsm_rolling.cpp and gsm_tangential.cpp.